src/gtimelog/timelog.py

```diff
@@ -1018,6 +1018,24 @@ def raw_append(self, line, need_space):
             f.write(line + '\n')
         self.last_mtime = get_mtime(self.filename)
 
+    def remove_last_entry(self):
+        self.check_reload()
+        if self.window.items:
+            self.window.items.pop()
+        else:
+            # entries list is empty, so nothing to remove
+            return None
+        _, last_entry = self.items.pop()
+        with open(self.filename, "r", encoding='utf-8') as f:
+            lines = f.readlines()
+        lines = lines[:-1]
+        if not lines[-1].strip():
+            # remove line which divides days if necessary
+            lines = lines[:-1]
```

Comment:
My greatest worry about this feature is that there's no undo if you (accidentally) press Ctrl+Backspace more than once in a row.  Maybe instead of removing the line from the file it would be better to comment it out?  Then at least you can recover with a text editor.

The timelog.txt format supports comments using `#` at the beginning of the line.  I plan to use some of the comments for directives  that set things like [timezones](https://github.com/gtimelog/gtimelog/issues/84#issuecomment-171241357) or virtual midnight times, so maybe we should use `##` for commented-out entries.

Reply:
I have been using this feature without undo for years in `gocept.gtimelog`. Yes, it happened to me that I called the command too often and thus loosing data. But I was usually able to reconstruct from memory, when it happened. (Mostly I use this feature when I added an entry because I thought I was done with a task, but then worked for some minutes on it to really complete it.)

So I see no immediate need for an undo ability. If it is too dangerous for some users, then they still can do it the old way be editing the timelog.txt file.

Reply:
The alternative could be, to do nothing when hitting the key combination and there is text in the field where you enter your time log entries.

Reply:
> The alternative could be, to do nothing when hitting the key combination and there is text in the field where you enter your time log entries.

I like this!